### PR TITLE
Create InboxPost model

### PR DESCRIPTION
**Pull request by debecerra**
### Summary
- Removed Inbox model, moved the friend_request, and posts collections to LocalAuthor
- Renamed Post to LocalPost
- Created InboxPost
- Posts that are created are sent out to the /inbox of all followers. The inbox posts are the posts that should be displayed on the feed of the user (currently only displayed in the "Inbox" tab)

I apologize for the **massive** amount of changes. These changes proved to be quite difficult to make due to a large amount of coupling between Posts and other aspects of the app, including likes, the home page, followers, etc. It was hard to leave the app in a working state. Most of the changes are contained in `socialDistribution/models/post.py` and the two `view.py` files. Most of the other file changes you can ignore as they are mostly model renames.

### Outstanding Issues
 1. Create Post form takes an image, but that image is never sent out due to the constraints of the Post API JSON specifications. Changes are needed to enforce that a Post is either just an image or markdown that may have an embedded image (URL to an image saved elsewhere). OR we create two posts if an author wants to attach an image, an Image Post and a Markdown Post.
 2. Author data for a post is not being fetched properly for inbox post authors. We need to find a new way to get the author info (such as display name) that involves API calls and possibly the Author model to cache it.
 3. Like counts that appear on the Post Cards are broken. These all relied on Django ORM functionality which now must be replaced by API calls. I left this code in quite a bad shape (i.e. just commented stuff out) because this would take quite some time to fix. This is probably the worst quality code that I'm trying to check in.
 4. Friends' posts are not working. We need to check if authors are friends via API calls, then send the post to does who are indeed friends. Right now, this uses Django ORM, which will not work. This is also related to the fact that followers as a whole are not fully implemented in the remote case.
 5. InboxPosts don't save all info that is sent to them. This was mostly to get a minimum possible version up and running as soon as possible.
 6. Editing and deleting posts is not reflected in the remote user since the post was already sent out. We need to implement update() method on the InboxPost model that does an API call to update it's fields, and deletes the InboxPost if the original post was deleted.
 7. Comment page don't work (will see if I can fix this before merge)
 ?. ... will add more if I remember anything else

As you can probably see, most of these issues are "Django ORM won't work anymore, need to set up API calls". I'd like to make this merge sooner rather than later so that I can get some help on these issues. This PR also has way too many changes already as-is. As a small bonus, all tests pass.

As a simple way to check: Create two users, follow one with the other, have a user make a public post. It will get sent to the inbox of the follower user. You can also send a user a post using the API, showing that the posts do not have to be local.

Screenshot from 2021-11-09 22-48-06

Notice everything that doesn't render properly on the "received" post, such as the author name.

Right now I'm going to focus on flushing out the handling of post authors and remote followers, as everything else will be dependent on this. Once these are working, we can focus on things getting likes for a post. I'll continue working on this and pushing "clean up" code while I wait for a review.

### CODE REVIEW

**Comment by debecerra** [comment]
Created #159, #160, #161, #162, #163 to address issues listed above.

**Comment by debecerra** [comment]
InboxPost UI now showing Author name

**Comment by debecerra** [comment]
This PR now resolves #160 by caching the data of the author when the post is received. This data does not update.

File: code/socialDistribution/templates/posts/comments.html [hunk 1]
```diff
@@ -35,50 +35,7 @@ <h2>Comments</h2>
...
-            $('#newCommentButton').prop('disabled', false);
-        } else {
-            $('#newCommentButton').prop('disabled', true);
-        }
-    })
-</script>
-=======
 {% load static %}
     <script src="{% static 'js/comment/comments.js' %}"></script>
->>>>>>> master
```
Comment (debecerra) [inline code/socialDistribution/templates/posts/comments.html:82]: Not sure why this was here? Seems an uncompleted merge was checked into git somehow

File: code/api/views.py [hunk 2]
```diff
@@ -298,22 +300,33 @@ def post(self, request, author_id):
...
-                inbox = get_object_or_404(Inbox, author_id=author_id)
-
-                # push post to inbox of author
-                try:
-                    post = Post.objects.get(id=post_id, author_id=post_author_id)
-                    inbox.posts.add(post)
-                except Post.DoesNotExist:
-                    raise ValueError()
+                # add post to inbox of author
+                receiving_author.inbox_posts.add(received_post)
```
Comment (debecerra) [inline code/api/views.py:329]: Inbox API handles remote posts

**Review by tylerelston — APPROVED** [review]
Nice work!

**Review by olivier-vadiaval — APPROVED** [review]
Great job! Sorry for the delay

File: code/socialDistribution/templates/posts/comments.html (hunk 1 above)
Comment (JawadHossain) [reply code/socialDistribution/templates/posts/comments.html:82]: O no. I hope no one force pushed

**Pull request merged**